mrp_production_project_estimated_cost/models/mrp_production.py

```diff
@@ -81,92 +87,129 @@ def action_compute(self, properties=None):
-                         line.workcenter_id.product_id.name))
-                vals = self._catch_information_estimated_cost(
-                    journal_wk, name, self, line,
-                    line.workcenter_id.product_id, op_number)
-                vals.update({
-                    'estim_average_cost': op_number * op_avg_cost,
-                    'estim_standard_cost': op_number * op_avg_cost,
-                })
+        id2 = self.env.ref(
+            'mrp_production_project_estimated_cost.estimated_cost_list_view')
+        search_view = self.env.ref('mrp_project_link.account_analytic_line'
+                                   '_mrp_search_view')
+        analytic_line_list = analytic_line_obj.search(
+            [('mrp_production_id', '=', self.id),
+             ('task_id', '=', False)])
+        self = self.with_context(search_default_group_production=1,
+                                 search_default_group_workorder=1,
+                                 search_default_group_journal=1)
+        return {
+            'view_type': 'form',
+            'view_mode': 'tree',
+            'res_model': 'account.analytic.line',
+            'views': [(id2.id, 'tree')],
+            'search_view_id': search_view.id,
+            'view_id': False,
+            'type': 'ir.actions.act_window',
+            'target': 'new',
+            'domain': "[('id','in',[" +
+            ','.join(map(str, analytic_line_list.ids)) + "])]",
+            'context': self.env.context
+            }
+
+    @api.multi
+    def calculate_production_estimated_cost(self):
+        analytic_line_obj = self.env['account.analytic.line']
+        for record in self:
+            cond = [('mrp_production_id', '=', record.id)]
+            analytic_line_obj.search(cond).unlink()
+            journal = record.env.ref('mrp_production_project_estimated_cost.'
+                                     'analytic_journal_materials', False)
+            for line in record.product_lines:
+                name = _('%s-%s' % (record.name, line.work_order.name or ''))
+                vals = record._prepare_estim_cost_analytic_line(
+                    journal, name, record, line.work_order, line.product_id,
+                    line.product_qty)
                 analytic_line_obj.create(vals)
+            journal = record.env.ref('mrp_production_project_estimated_cost.'
+                                     'analytic_journal_machines', False)
+            for line in record.workcenter_lines:
+                op_wc_lines = line.routing_wc_line.op_wc_lines
+                wc = op_wc_lines.filtered(lambda r: r.workcenter ==
+                                          line.workcenter_id) or \
+                    line.workcenter_id
+                time_start = wc.time_start
+                time_stop = wc.time_stop
+                op_number = wc.op_number
+                op_avg_cost = wc.op_avg_cost
+                if (time_start and line.workcenter_id.pre_op_product):
```

Comment:
Ana, acabo de tener un déjà vú con tu comentario de otro momento en el que hablábamos de exactamente lo mismo con esto de los costes...

Pero bueno, que el raise es correcto siempre que el proceso sea interactivo y deba ser bloqueante, y esto cumple ambos. No dejar confirmar la OF si eso no lo tienes bien puesto. En el _prepare es buen sitio, pero claro, no va a llegar a ese prepare si en el otro lado le tienes puesto un `if line.cycle and line.workcenter_id.product_id:`